pkgs/servers/irc/solanum/default.nix

```diff
@@ -0,0 +1,57 @@
+{ stdenv
+, fetchFromGitHub
+, autoreconfHook
+, pkg-config
+, bison
+, flex
+, openssl
+, sqlite
+, lksctp-tools
+}:
+
+stdenv.mkDerivation rec {
+  pname = "solanum";
+  version = "unstable-2020-12-14";
+
+  src = fetchFromGitHub {
+    owner = "solanum-ircd";
+    repo = pname;
+    rev = "551e5a146eab4948ce4a57d87a7f671f2d7cc02d";
+    sha256 = "14cd2cb04w6nwck7q49jw5zvifkzhkmblwhjfskc2nxcdb5x3l96";
+  };
+
+  configureFlags = [
+    "--enable-epoll"
+    "--enable-ipv6"
+    "--enable-openssl=${openssl.dev}"
+    "--enable-sctp=${lksctp-tools.out}/lib"
+    "--with-program-prefix=solanum-"
+    "--localstatedir=/var/lib/solanum"
+  ];
+
+  nativeBuildInputs = [
+    autoreconfHook
+    bison
+    flex
+    pkg-config
+  ];
+
+  buildInputs = [
+    openssl
+    sqlite
+  ];
+
+  doCheck = true;
+  checkPhase = ''
+    make check
```

Comment:
`doCheck` should be sufficient as the default checkPhase already tries both `make test` and `make check`.